api/server/http-rest/server.go

```diff
@@ -0,0 +1,112 @@
+package http_rest
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v5/api/server/middleware"
+	"github.com/prysmaticlabs/prysm/v5/runtime"
+)
+
+var _ runtime.Service = (*Server)(nil)
+
+// restHandler is a functional interface that implements the rest handler functionality.
+type restHandler func(
+	h http.HandlerFunc,
+	w http.ResponseWriter,
+	req *http.Request,
+)
+
+// Config parameters for setting up the http-rest service.
+type config struct {
+	httpAddr       string
+	allowedOrigins []string
+	muxHandler     restHandler
+	router         *mux.Router
+	timeout        time.Duration
+}
+
+// Server serves HTTP JSON traffic.
+type Server struct {
+	cfg          *config
+	server       *http.Server
+	cancel       context.CancelFunc
+	ctx          context.Context
+	startFailure error
+}
+
+// New returns a new instance of the Server.
+func New(ctx context.Context, opts ...Option) (*Server, error) {
```

Comment:
What is this package about? seems unrelated to removing gRPC gateway

Reply:
this is the rename/rewrite of api/gateway/gateway.go the rest server still needs to run

Reply:
These are all showing up as new additions, that was my confusion. Is it too difficult to go back and move these files via `git mv` so that they are linked as a rename/move rather than an entirely new package? That would make it easier to review the diff

Reply:
need to double check, might be difficult to do that, I didn't realize git mv did that 😓 
the contents of the gateway package got moved to this http-rest package, and the internals removed. I think that's the main difference, can attempt to do it later today

Reply:
i tried to git mv but didn't work out correctly

I created a temp branch from this one
reverted the branch to develop ( maybe I should have force pushed?)
moved the files, committed
merged the temp branch back to this one 
pushed. 

it didn't seem to work.

Reply:
I opened up a second window with the PR and just compared them side by side